### DISCUSSION TITLE
Preview function for "Find and replace"

### DISCUSSION THREAD

**Discussion by nitech** [craftcms/cms#8720] [Ideas]
It would be nice to know which entries are affected by a "Find and replace". Right now it's kind of "fire and forget" and feels a bit risky to initiate a "Find and replace".

**Comment by sebastian-lenz** [comment]
I actually always wondered you is using this function. Generally seems quite dangerous to me.

**Comment by BeKnowDo** [comment]
> I actually always wondered you is using this function. Generally seems quite dangerous to me.

EXTREMELY dangerous

**Comment by GaryReckard** [comment]
Yeah, I feel like a dry-run option would be very useful. Basically, if it's running some sort of 

`UPDATE table SET field = REPLACE(field, 'New Text', 'Old Text') where field_caption LIKE BINARY '%Old Text%';`

to perform the find&replace, perhaps the Dry Run could just run something like 

`select * from table where field LIKE BINARY '%Old Text%';`

and show you all the values that would get changed, with some surrounding context...?
